### PR TITLE
Test: recover search index tool it in multi node cluster

### PR DESCRIPTION
**Pull request by yuye-aws**
### Description
Recover search index tool it in multi node cluster 

### Issues Resolved
Issue: https://github.com/opensearch-project/ml-commons/issues/2362
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ylwu-amzn** [comment]
The test failed `org.opensearch.ml.tools.VisualizationsToolIT.testVisualizationNotFound`, seems flaky. Let's wait for Zhichao's PR merge and rebase ?

**Comment by yuye-aws** [comment]
@ylwu-amzn There is no conflict for this PR now, you can review first.

**Comment by yuye-aws** [comment]
@ylwu-amzn If we are merging this PR, please attach backport 2.x label.

**Comment by yuye-aws** [comment]
> The test failed `org.opensearch.ml.tools.VisualizationsToolIT.testVisualizationNotFound`, seems flaky. Let's wait for Zhichao's PR merge and rebase ?
> 
> Update: Zhichao's PR #2416 merged. You can rebase now

Actually it's hailong's PR

File: plugin/src/test/java/org/opensearch/ml/rest/RestSearchIndexToolIT.java
Comment (ylwu-amzn) [inline plugin/src/test/java/org/opensearch/ml/rest/RestSearchIndexToolIT.java:113]: Can you try `getDetailedMessage` method  of `RemoteTransportException`  ? I think it could get detailed message

Comment (yuye-aws) [reply plugin/src/test/java/org/opensearch/ml/rest/RestSearchIndexToolIT.java:113]: Unfortunately, `getDetailedMessage` method is only available for `OpenSearchException`. Here, the exception is always of type `ResponseException` (even in the multi cluster test).

Comment (yuye-aws) [reply plugin/src/test/java/org/opensearch/ml/rest/RestSearchIndexToolIT.java:113]: I will update the comment to avoid confusing people.

File: plugin/src/test/java/org/opensearch/ml/rest/RestSearchIndexToolIT.java
Comment (ylwu-amzn) [inline plugin/src/test/java/org/opensearch/ml/rest/RestSearchIndexToolIT.java:114]: I think this is related with https://github.com/opensearch-project/ml-commons/issues/2443

@jngz-es, can you help fix the bug ? Then we can ask @yuye-aws to verify if this error is gone on multi-node cluster

Comment (yuye-aws) [reply plugin/src/test/java/org/opensearch/ml/rest/RestSearchIndexToolIT.java:114]: Thanks @jngz-es for creating this issue! We can wait merging this PR before https://github.com/opensearch-project/ml-commons/issues/2443

**Comment by yuye-aws** [comment]
Just see that https://github.com/opensearch-project/ml-commons/pull/2448 has been merged. I am taking a look at this issue.

**Comment by yuye-aws** [comment]
The CI checks failed due to flakey tests. We can rerun the tests after merging this PR.

**Comment by yuye-aws** [comment]
Hi maintainers, can you help attack backport 2.x, backport 2.14 and backport 2.15 labels to this PR?

**Comment by yuye-aws** [comment]
> The CI checks failed due to flakey tests. We can rerun the tests after merging this PR: #2437

Since this PR has been merged. Can you help rerun the tests?

File: plugin/build.gradle
Comment (dhrubo-os) [inline plugin/build.gradle:197]: Looks like there's a note why this section is commented out. How are we solving this issue then?

Comment (yuye-aws) [reply plugin/build.gradle:197]: We are bringing back IT search index tool for multi node cluster.

Comment (yuye-aws) [reply plugin/build.gradle:197]: After merging https://github.com/opensearch-project/ml-commons/issues/2443 the issue has been resolved.

**Comment by yuye-aws** [comment]
Can anyone help rerun the tests? I don't think the current CI failure is due to this PR.

**Comment by yuye-aws** [comment]
Rerun tests?

**Comment by yuye-aws** [comment]
Can you guys please review this PR? @ylwu-amzn @jngz-es @dhrubo-os

**Comment by mingshl** [comment]
LGTM, the CI are all passing now. Thanks for bring back this IT

**Review by mingshl — APPROVED** [review]

**Comment by yuye-aws** [comment]
@mingshl Please add backport 2.x label to this PR, thanks

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**
